### PR TITLE
Admin pages

### PR DESCRIPTION
**Pull request by SJD401**
Add Admin pages  in a new branch, if it was acceptable, merge it .

### CODE REVIEW

File: projectsmember/Login_User.aspx.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+namespace projectsmember
+{
+    public partial class Login_User : System.Web.UI.Page
+    {
+        SqlConnection LOGConn = new SqlConnection(ConfigurationManager.ConnectionStrings["sojaIrishConnectionString"].ConnectionString);
+
+        protected void Page_Load(object sender, EventArgs e)
+        {
+            txtUserName.Attributes.Add("placeholder", "Username");
```
Comment (soja-soja) [inline projectsmember/Login_User.aspx.cs:19]: Why don't we set these in client side? In html...
https://www.w3schools.com/tags/att_input_placeholder.asp

File: projectsmember/Project Manager.aspx.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            }
+
+
+            if (LOGConn.State == ConnectionState.Open)
+            {
+                LOGConn.Close();
+            }
+            LOGConn.Open();
+
+            DataSet1TableAdapters.ProjectsTableAdapter dstaProjNotif2 = new DataSet1TableAdapters.ProjectsTableAdapter();
```
Comment (soja-soja) [inline projectsmember/Project Manager.aspx.cs:36]: Ok, so here, we don't need LOGConn at all. 

we should be able to do those all with dataset object. You need to define a new query there, where you pass username and password and the rest is taken care of.  Just like how I did it in this Video: 

https://www.youtube.com/watch?v=0NnwpeqEFOs&ab_channel=SOJA

**Review by soja-soja — CHANGES_REQUESTED** [review]
Left a number of comments - please address them.

File: projectsmember/Login_User.aspx.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                }
+            }
+
+            DataSet1TableAdapters.MembersTableAdapter dstaMem = new DataSet1TableAdapters.MembersTableAdapter();
+            int numberOfUsers = dstaMem.QueryLoginUser(txtUserName.Text, txtPass.Text).GetHashCode();
+
+            if (numberOfUsers > 0)
+            {
+                //Login successful !
+                Session.Add("status", "Login_User");
```
Comment (soja-soja) [inline projectsmember/Login_User.aspx.cs:52]: someone can just set the same session values and pretend that is already logged in. We need to do the followings here: 

if the user is successfully passed the username/password checking (authentication) then we generate a random string (token), and saved it somewhere in our database (maybe user table in an additional column for each user) and then return and save it in the session. 


then in the next pages, we say if what we have in session matches what we have in database (user doesn't need to do anything here) we assume the user is logged in. 

on each successful login, we generate a new random token ...

File: projectsmember/Profile.aspx.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+namespace projectsmember
+{
+    public partial class Profile : System.Web.UI.Page
+    {
+        protected void Page_Load(object sender, EventArgs e)
+        {
+
+
+            if (Session["status"] != null &&
+                Session["status"].ToString() == "Login_User")
```
Comment (soja-soja) [inline projectsmember/Profile.aspx.cs:17]: as mentioned in earlier comment, if I set the "status" variable in my browser and the value to the pre-defined "Login_User" , I can be the admin of this website! note that the code is also public, so anyone can know about these pre-defined values.

File: projectsmember/Project Manager.aspx.cs (hunk 2 above)
Comment (SJD401) [reply projectsmember/Project Manager.aspx.cs:36]: Was corrected

File: projectsmember/Login_User.aspx.cs (hunk 1 above)
Comment (SJD401) [reply projectsmember/Login_User.aspx.cs:19]: You're Right. Resolved

File: projectsmember/Profile.aspx.cs (hunk 4 above)
Comment (SJD401) [reply projectsmember/Profile.aspx.cs]: Done !

File: projectsmember/Login_User.aspx.cs (hunk 3 above)
Comment (SJD401) [reply projectsmember/Login_User.aspx.cs]: Done, thanks for your help

**Pull request merged**